infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -420,6 +426,10 @@ protected void startMachines() throws InfrastructureException {
     // namespace.pods().watch(new AbnormalStopHandler());
     // namespace.pods().watchContainers(new MachineLogsPublisher());
 
+    if (failFastOnUnrecoverableEvents) {
+      namespace.pods().watchContainers(new UnrecoverableEventHanler());
```

Comment:
I think it could be useful add `MachineLogsPublisher` here. WDYT?

Reply:
@sleshchenko can you elaborate on what exactly to do?

Reply:
I think it should be something like the following:
```
--> start
try {
  namespace.pods().watchContainers(new MachineLogsPublisher());
  if (failFastOnUnrecoverableEvents) {
    namespace.pods().watchContainers(new UnrecoverableEventHanler());
  }

  //doStart here
} catch ....
... 
} finally {
  namespace.pods().stopWatch();
}
```
In this case clients would receive K8s/OS events text as machine's output, and it would be easier to understand why workspace start fails.

Reply:
@sleshchenko Adding `MachineLogsPublisher` seems to be unrelated to this issue but rather to https://github.com/eclipse/che/issues/7653 IMO. Could this be addressed in another PR ?

Reply:
@ibuziuk For me, it looks like it's easy to add it here (I guess described changes should be enough) and it will be useful (machine logs panel will have all events). It's why I write it here. 
But I agree that it is not related to a solved issue and it can be added in another PR.

Reply:
@sleshchenko ok, sounds relevant - let me add it to this PR